server/server.go

```diff
@@ -270,10 +268,10 @@ func newServer(ctx context.Context, c Config, rotationStrategy rotationStrategy)
 		}
 
 		instrumentHandlerCounter = func(handlerName string, handler http.Handler) http.HandlerFunc {
```

Comment:
Not the scope of this PR, but I wonder if this should return an `http.Handler` instead.

Reply:
I think it returns http.HandlerFunc because of the way `httpsnoop` works. For me, it looks better to fix it in another PR with server refactoring.